liquibase-standard/src/main/java/liquibase/integration/spring/UnexpectedChangeSetsCustomizer.java

```diff
@@ -0,0 +1,79 @@
+package liquibase.integration.spring;
+
+import liquibase.Contexts;
+import liquibase.LabelExpression;
+import liquibase.Liquibase;
+import liquibase.command.core.UnexpectedChangesetsCommandStep;
+import liquibase.exception.LiquibaseException;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+
+public class UnexpectedChangeSetsCustomizer<T extends Liquibase> implements Customizer<T>, ApplicationListener<ContextRefreshedEvent> {
+
+    private final Set<ChangeSetInfo> applied = new HashSet<>(), unexpected = new HashSet<>();
+
+    private final Consumer<Set<ChangeSetInfo>> unexpectedChangeSetsConsumer;
+
+    public UnexpectedChangeSetsCustomizer(Consumer<Set<ChangeSetInfo>> unexpectedChangeSetsConsumer) {
+        this.unexpectedChangeSetsConsumer = unexpectedChangeSetsConsumer;
+    }
+
+    @Override
+    public void customize(T liquibase) {
+        try {
+            liquibase.getDatabaseChangeLog().getChangeSets().stream().map(changeSet -> new ChangeSetInfo(
+                    changeSet.getId(),
+                    changeSet.getChangeLog().getFilePath(),
+                    changeSet.getAuthor())).forEach(applied::add);
+            UnexpectedChangesetsCommandStep.listUnexpectedChangeSets(
+                liquibase.getDatabase(),
+                liquibase.getDatabaseChangeLog(),
+                new Contexts(),
```

Comment:
The contexts and labels are available in [SpringLiquibase ](https://github.com/liquibase/liquibase/blob/master/liquibase-standard/src/main/java/liquibase/integration/spring/SpringLiquibase.java#L76) class.  It there a way to get them here and use the contexts and labels from it?

Reply:
If the customizer is registered as a bean in Spring, it does not receive the `SpringLiquibase` but only the `Liquibase` instance. But one could add a method to adjust.

Reply:
I added a way to add `LabelExpressions` and `Contexts`.

Reply:
Hi @raphw - did you mean to push changes to add labels and contexts?

Reply:
I did but tried to push to origin and not to my branch, so it failed. Fixed now!